### PR TITLE
[Notifications] Feat: Extensions notifications

### PR DESCRIPTION
**Pull request by iamsamgibbs**
[See CDapp counterpart](https://github.com/JoinColony/colonyCDapp/pull/3290)

(Note: Also updated package-lock.json as it was breaking CDapp, but should probably be done on the feat/in-app-notifications branch)

### CODE REVIEW

**Review by rdig — APPROVED** [review]
Tested as part of the review of JoinColony/colonyCDapp#3290

File: src/utils/notifications.ts [hunk 1]
```diff
@@ -200,6 +223,32 @@ export const sendExpenditureUpdateNotifications = async ({
...
+
+  if (!recipients.length) {
+    return;
+  }
+
+  // Send the colony wide notifications.
+  await sendNotification(`Extension: ${extensionHash}`, recipients, {
+    notificationType,
+    creator,
+    notificationCategory: NotificationCategory.Extension,
```
Comment (bassgeta) [inline src/utils/notifications.ts:245]: cc @davecreaser should we use extension as a category or are these product wise included in the `Admin` category?

**Review by bassgeta — APPROVED** [review]
Love the new `updateExtension` helper, it gets the job done well.
And naturally, the entire solution is also very sensible and nice :sunglasses: 
Tested as part of the CDapp :+1:

File: src/utils/notifications.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ export enum NotificationCategory {
   Mention = 'Mention',
   Payment = 'Payment',
   Admin = 'Admin',
+  Extension = 'Extension',
```
Comment (davecreaser) [inline src/utils/notifications.ts:19]: The categories should be limited to only Mention, Payment and Admin, so all the extension notifications can be under Admin category. Sorry, I should really have left a comment about this when making this type.

File: src/utils/notifications.ts (hunk 1 above)
Comment (davecreaser) [reply src/utils/notifications.ts:245]: Yeah you're correct, they should all use Admin as the category. My fault really for not making that clear in tickets / with comments in the code.

Comment (davecreaser) [reply src/utils/notifications.ts:245]: PS. good catch!

**Review by rumzledz — APPROVED** [review]

**Pull request merged**